### PR TITLE
Fix `TexturePreview` crashing

### LINKED ISSUE #50702: Crash in TexturePreview when creating a new Resource and selecting StreamTexture
### Godot version

3.x (40b57319e23856a0cba54cbeeda6bba4cc29b589), master (a0aeb564244247996a5509be568c881d560fab77)

### System information

Linux, Mageia 9 x86_64

### Issue description

When attempting to create a new Resource from the FileSystem dock, and selecting `StreamTexture` in the create dialog (with a double click), the editor crashes (on `3.x`).

Given the stacktrace, I would assume it's related to cf8de3b316f35ed296b6fb51fd6b4ad643bda66f, CC @LightningAA.

```
ERROR: Index p_format = 37 is out of bounds (FORMAT_MAX = 37).
   at: get_format_name (core/image.cpp:3226)
handle_crash: Program crashed with signal 11
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] /lib64/libc.so.6(+0x3ac90) [0x7f8cff698c90] (??:0)
[2] DynamicFont::set_size(int) (/home/akien/Projects/godot/godot-3.x/scene/resources/dynamic_font.cpp:716)
[3] TexturePreview::TexturePreview(Ref<Texture>, bool) (/home/akien/Projects/godot/godot-3.x/editor/plugins/texture_editor_plugin.cpp:70)
[4] EditorInspectorPluginTexture::parse_begin(Object*) (/home/akien/Projects/godot/godot-3.x/editor/plugins/texture_editor_plugin.cpp:93 (discriminator 1))
[5] EditorInspector::update_tree() (/home/akien/Projects/godot/godot-3.x/editor/editor_inspector.cpp:1453)
[6] EditorInspector::edit(Object*) (/home/akien/Projects/godot/godot-3.x/editor/editor_inspector.cpp:1832)
[7] EditorNode::_edit_current() (/home/akien/Projects/godot/godot-3.x/editor/editor_node.cpp:1978)
[8] EditorNode::push_item(Object*, String const&, bool) (/home/akien/Projects/godot/godot-3.x/editor/editor_node.cpp:1907)
[9] FileSystemDock::_resource_created() const (/home/akien/Projects/godot/godot-3.x/editor/filesystem_dock.cpp:1940 (discriminator 3))
[10] MethodBind0C::call(Object*, Variant const**, int, Variant::CallError&) (/home/akien/Projects/godot/godot-3.x/./core/method_bind.gen.inc:423 (discriminator 4))
[11] Object::call(StringName const&, Variant const**, int, Variant::CallError&) (/home/akien/Projects/godot/godot-3.x/core/object.cpp:918 (discriminator 1))
[12] Object::emit_signal(StringName const&, Variant const**, int) (/home/akien/Projects/godot/godot-3.x/core/object.cpp:1226 (discriminator 1))
[13] Object::emit_signal(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (/home/akien/Projects/godot/godot-3.x/core/object.cpp:1282)
[14] CreateDialog::_confirmed() (/home/akien/Projects/godot/godot-3.x/editor/create_dialog.cpp:455 (discriminator 2))
[15] MethodBind0::call(Object*, Variant const**, int, Variant::CallError&) (/home/akien/Projects/godot/godot-3.x/./core/method_bind.gen.inc:61 (discriminator 4))
[16] Object::call(StringName const&, Variant const**, int, Variant::CallError&) (/home/akien/Projects/godot/godot-3.x/core/object.cpp:918 (discriminator 1))
[17] Object::emit_signal(StringName const&, Variant const**, int) (/home/akien/Projects/godot/godot-3.x/core/object.cpp:1226 (discriminator 1))
[18] Object::emit_signal(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (/home/akien/Projects/godot/godot-3.x/core/object.cpp:1282)
[19] Tree::_gui_input(Ref<InputEvent>) (/home/akien/Projects/godot/godot-3.x/scene/gui/tree.cpp:2617 (discriminator 2))
[20] MethodBind1<Ref<InputEvent> >::call(Object*, Variant const**, int, Variant::CallError&) (/home/akien/Projects/godot/godot-3.x/./core/method_bind.gen.inc:793 (discriminator 12))
[21] Object::call_multilevel(StringName const&, Variant const**, int) (/home/akien/Projects/godot/godot-3.x/core/object.cpp:766 (discriminator 1))
[22] Object::call_multilevel(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (/home/akien/Projects/godot/godot-3.x/core/object.cpp:864)
[23] Viewport::_gui_call_input(Control*, Ref<InputEvent> const&) (/home/akien/Projects/godot/godot-3.x/scene/main/viewport.cpp:1632 (discriminator 2))
[24] Viewport::_gui_input_event(Ref<InputEvent>) (/home/akien/Projects/godot/godot-3.x/scene/main/viewport.cpp:1951 (discriminator 3))
[25] Viewport::input(Ref<InputEvent> const&) (/home/akien/Projects/godot/godot-3.x/scene/main/viewport.cpp:2783 (discriminator 2))
[26] Viewport::_vp_input(Ref<InputEvent> const&) (/home/akien/Projects/godot/godot-3.x/scene/main/viewport.cpp:1402)
[27] MethodBind1<Ref<InputEvent> const&>::call(Object*, Variant const**, int, Variant::CallError&) (/home/akien/Projects/godot/godot-3.x/./core/method_bind.gen.inc:793 (discriminator 12))
[28] Object::call(StringName const&, Variant const**, int, Variant::CallError&) (/home/akien/Projects/godot/godot-3.x/core/object.cpp:918 (discriminator 1))
[29] Object::call(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (/home/akien/Projects/godot/godot-3.x/core/object.cpp:848)
[30] SceneTree::call_group_flags(unsigned int, StringName const&, StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (/home/akien/Projects/godot/godot-3.x/scene/main/scene_tree.cpp:270)
[31] SceneTree::input_event(Ref<InputEvent> const&) (/home/akien/Projects/godot/godot-3.x/scene/main/scene_tree.cpp:429 (discriminator 6))
[32] InputDefault::_parse_input_event_impl(Ref<InputEvent> const&, bool) (/home/akien/Projects/godot/godot-3.x/main/input_default.cpp:437)
[33] InputDefault::parse_input_event(Ref<InputEvent> const&) (/home/akien/Projects/godot/godot-3.x/main/input_default.cpp:260)
[34] InputDefault::flush_accumulated_events() (/home/akien/Projects/godot/godot-3.x/main/input_default.cpp:653)
[35] OS_X11::process_xevents() (/home/akien/Projects/godot/godot-3.x/platform/x11/os_x11.cpp:2906)
[36] OS_X11::run() (/home/akien/Projects/godot/godot-3.x/platform/x11/os_x11.cpp:3636)
[37] godot-3.x(main+0x109) [0x177182f] (/home/akien/Projects/godot/godot-3.x/platform/x11/godot_x11.cpp:57)
[38] /lib64/libc.so.6(__libc_start_main+0xcd) [0x7f8cff6857dd] (??:0)
[39] godot-3.x(_start+0x2a) [0x177167a] (??:?)
-- END OF BACKTRACE --
Aborted (core dumped)
```

On `master`, it doesn't seem to crash, but the error message is raised too:
```
ERROR: Index p_format = 39 is out of bounds (FORMAT_MAX = 39).
   at: get_format_name (core/io/image.cpp:3499)
```

### Steps to reproduce

- Right click on FileSystem dock, select New Resource
- Double click on StreamTexture
- Crash

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by AaronRecord**
Fixes https://github.com/godotengine/godot/issues/50702 for `master`

### CODE REVIEW

File: editor/plugins/texture_editor_plugin.cpp [hunk 1]
```diff
@@ -36,9 +36,24 @@ TextureRect *TexturePreview::get_texture_display() {
 	return texture_display;
 }
 
+void TexturePreview::_notification(int p_what) {
+	switch (p_what) {
+		// TODO: `case NOTIFICATION_THEME_CHANGED:` should probably be here as well,
+		// but `NOTIFICATION_THEME_CHANGED` is for some reason getting called when
+		// the `TexturePreview` is getting destroyed, which causes `get_theme_font()`
+		// to return `nullptr`.
+		// See https://github.com/godotengine/godot/issues/50743.
```
Comment (akien-mga) [inline editor/plugins/texture_editor_plugin.cpp:45]: You can still handle it and use `if (is_instance_tree) {}`, that's a common pattern to work the issue around. See e.g. `EditorHelp`.

Comment (LightningAA) [reply editor/plugins/texture_editor_plugin.cpp:45]: Alright, I'll do that.

Comment (LightningAA) [reply editor/plugins/texture_editor_plugin.cpp]: Done.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**